game-core/src/main/java/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -61,6 +65,113 @@ public UnitImageFactory(
     this.mapData = mapData;
   }
 
+  @Value
+  @Builder
+  public static class ImageKey {
+    private final GamePlayer player;
+    private final UnitType type;
+    private final boolean damaged;
+    private final boolean disabled;
+
+    public static ImageKey of(final UnitCategory unit) {
+      return ImageKey.builder()
+          .player(unit.getOwner())
+          .type(unit.getType())
+          .damaged(unit.hasDamageOrBombingUnitDamage())
+          .disabled(unit.getDisabled())
+          .build();
+    }
+
+    public static ImageKey of(final UnitOwner holder) {
+      return ImageKey.builder().player(holder.getOwner()).type(holder.getType()).build();
+    }
+
+    public static ImageKey of(final Unit unit) {
+      return ImageKey.builder()
+          .player(unit.getOwner())
+          .type(unit.getType())
+          .damaged(Matches.unitHasTakenSomeBombingUnitDamage().test(unit))
+          .disabled(Matches.unitIsDisabled().test(unit))
+          .build();
+    }
+
+    public String getFullName() {
+      return getBaseImageName() + player.getName();
+    }
+
+    public String getBaseImageName() {
```

Comment:
Codacy found an issue: [The method 'getBaseImageName()' has an NPath complexity of 102606, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5910459)